lib/dfe/analytics/entity_table_check_job.rb

```diff
@@ -26,17 +26,54 @@ def perform
       end
 
       def entity_table_check_data(model)
+        adapter_name = ActiveRecord::Base.connection.adapter_name.downcase
+
+        if adapter_name != 'postgresql' && Rails.env.production?
+          Rails.logger.info('DfE::Analytics: Entity checksum: Only Postgres databases supported on PRODUCTION')
+          return
+        end
+
+        row_count, checksum, checksum_calculated_at = fetch_checksum_data(model, adapter_name)
+        {
+          row_count: row_count,
+          checksum: checksum,
+          checksum_calculated_at: checksum_calculated_at
+        }
+      end
+
+      def fetch_checksum_data(model, adapter_name)
+        if adapter_name == 'postgresql'
+          fetch_postgresql_checksum_data(model)
+        else
+          fetch_generic_checksum_data(model)
+        end
+      end
+
+      def fetch_postgresql_checksum_data(model)
+        checksum_sql_query = <<-SQL
+          SELECT COUNT(*) as row_count,
+            MD5(STRING_AGG(CHECKSUM_TABLE.ID, '')) as checksum,
+            CURRENT_TIMESTAMP(6) as checksum_calculated_at
```

Comment:
Will the `CURRENT_TIMESTAMP(6)` here be the same as `CURRENT_TIMESTAMP(6)` in the where clause. I thought it possibly could be different if the query takes a long time to run.

Reply:
@asatwal - My thought process was that this SELECT would be transactional and so the timestamp would be the same throughout, but your suggestion might be a better way. Maybe @stevenleggdfe might have some insight?

Reply:
@asatwal @ericaporter what would you think about ```MAX(updated_at) AS checksum_calculated_at``` instead? Or would that hit performance too heavily?

Reply:
It looks like `MAX(updated_at)` would be much more accurate than getting the timestamp at the beginning of the calculation (my suggestion), but I guess could potentially cause performance issues. I am wondering if @asatwal's suggestion might be somewhere in between these two where trade offs are concerned?

Reply:
updated_at is not an indexed column on most database tables, so this may well cause performance problems on large tables. I think we need to avoid any chance of performance problems after our last experience.

I don't believe it will make much of difference to accuracy as once we have a timestamp and any transactions updated after this timestamp will be excluded, both in our calculations and the BQ End.  Can't think of a scenario where MAX(UPDATED_AT) would be more accurate, unless I'm not thinking hard enough :-)

Reply:
I guess I'm thinking real edge cases like if some update is processing in parallel and hasn't been written to the db yet. Not sure if that's even possible though or whether that would cause a problem anyway. So if it creates performance issues let's just stick with current_timestamp(6) I think.

Reply:
> I guess I'm thinking real edge cases like if some update is processing in parallel and hasn't been written to the db yet. Not sure if that's even possible though or whether that would cause a problem anyway. So if it creates performance issues let's just stick with current_timestamp(6) I think.

Indeed, maybe a possibility that you've taken the `current_timestamp` and a long parallel process that was running before you took the timestamp finishes and writes to the database just as you're calculating the checksum. Even in that case the `updated_at` should be slightly later than your `current_timestamp` assuming that `updated_at` is calculated on the write to the database and not before. I think it's a risk we can take for now.